crypto/evp/digest.c

```diff
@@ -1078,7 +1078,13 @@ static void evp_md_free(void *md)
 EVP_MD *EVP_MD_fetch(OSSL_LIB_CTX *ctx, const char *algorithm,
     const char *properties)
 {
-    EVP_MD *md = evp_generic_fetch(ctx, OSSL_OP_DIGEST, algorithm, properties,
+    EVP_MD *md = NULL;
+
+    if (evp_generic_fetch_frozen(ctx, OSSL_OP_DIGEST, algorithm, properties,
+            NULL, (void **)&md))
+        return md;
+
+    md = evp_generic_fetch(ctx, OSSL_OP_DIGEST, algorithm, properties,
```

Comment:
Presumably, there will be subsequent PRs that make the same modification for EVP_CIPHER, EVP_KEM, EVP_KDF, etc, right?

If so, why make the same modification at every call site of evp_generic_fetch?  Why not call evp_generic_fetch_frozen from inside evp_generic_fetch, and fall back to the non-frozen behavior if fetch_frozen returns zero?  That way (I think) you get frozen method store benefits for all our EVP types at once, rather than having to make this shift above for every call site?

Reply:
We considered that, but we don't think that it'll work for us because of how the frozen cache will be implemented (one per EVP type). If we can make it generic, then we will. You can see how we want it work in this issue: https://github.com/openssl/project/issues/1723, but the actual implementation will be done in https://github.com/openssl/openssl/pull/29433